klarna-on-demand/src/test/java/com/klarna/ondemand/crypto/SharedPreferencesCryptoImplTest.java

```diff
@@ -27,61 +23,30 @@
 
 @RunWith(RobolectricTestRunner.class)
 @Config(emulateSdk = 18)
-@PrepareForTest(CryptoSharedPreferencesBaseImpl.class)
+@PrepareForTest(SharedPreferencesCryptoImpl.class)
 @PowerMockIgnore({ "org.mockito.*", "org.robolectric.*", "android.*", "org.json.*" })
-public class CryptoSharedPreferencesBaseImplTest {
-
-    private Crypto crypto;
-
-    @Before
-    public void beforeEach() {
-        android.content.Context context = Robolectric.application.getApplicationContext();
-        crypto = CryptoSharedPreferencesBaseImpl.getInstance(context);
-    }
-
-    @Test
-    public void getPublicKeyBase64Str_shouldReturnSameKeyOnTwoConsecutiveCalls () {
-        String publicKeyA = crypto.getPublicKeyBase64Str();
-        String publicKeyB = crypto.getPublicKeyBase64Str();
+public class SharedPreferencesCryptoImplTest extends CryptoBaseTst {
 
-        assertThat(publicKeyA)
-                .isNotEmpty()
-                .isEqualTo(publicKeyB);
+    @Override
+    protected int getTestSdkVersion() {
+        return 17;
     }
 
-    //region #sign
-    @Test
-    public void sign_shouldReturnSameSignOnTwoIdenticalCalls() throws NoSuchAlgorithmException, InvalidKeyException, SignatureException {
-        String signA = crypto.sign("my_messge");
-        String signB = crypto.sign("my_messge");
-
-        assertThat(signA)
-                .isNotEmpty()
-                .isEqualTo(signB);
-    }
-
-    @Test
-    public void sign_shouldReturnDifferentSignOnTwoDifferentCalls() throws NoSuchAlgorithmException, InvalidKeyException, SignatureException {
-        String signA = crypto.sign("my_messgeA");
-        String signB = crypto.sign("my_messgeB");
-
-        assertThat(signA)
-                .isNotEmpty()
-                .isNotEqualTo(signB);
-
-        assertThat(signB).isNotEmpty();
+    @Override
+    protected Crypto getTestSubject(Context context) throws Exception {
+        return new SharedPreferencesCryptoImpl(context);
     }
 
     @Test
     public void sign_shouldCaclculateAValidSignature() throws NoSuchAlgorithmException, InvalidKeyException, SignatureException, InvalidKeySpecException {
```

Comment:
Right. Might be worth adding a comment about that, so that possible contributors with a penchant for tidying up don't try something like that out without noticing this particular problem.